internal/converter/internal/staticconvert/internal/build/builder_traces.go

```diff
@@ -68,6 +72,82 @@ func (b *ConfigBuilder) translateAutomaticLogging(otelCfg *otelcol.Config, cfg t
 	removeProcessor(otelCfg, "traces", "automatic_logging")
 }
 
+func (b *ConfigBuilder) translateSpanMetrics(otelCfg *otelcol.Config, cfg traces.InstanceConfig) {
+	if _, ok := otelCfg.Processors[otel_component.NewID("spanmetrics")]; !ok {
+		return
+	}
+
+	// Remove the custom otel components and delete the custom pipeline
+	removeProcessor(otelCfg, "traces", "spanmetrics")
+	removeReceiver(otelCfg, "metrics", "noop")
+	removeExporter(otelCfg, "metrics", "remote_write")
+	removeExporter(otelCfg, "metrics", "prometheus")
+	delete(otelCfg.Service.Pipelines, otel_component.NewIDWithName("metrics", "spanmetrics"))
+
+	// If the spanmetrics configuration includes a handler_endpoint, we cannot convert it.
+	// This is intentionally after the section above which removes the custom spanmetrics processor
+	// so that the rest of the configuration can optionally be converted with the error.
+	if cfg.SpanMetrics.HandlerEndpoint != "" {
+		b.diags.Add(diag.SeverityLevelError, "Cannot convert using configuration including spanmetrics handler_endpoint. "+
+			"No equivalent exists for exposing a known /metrics endpoint. You can use metrics_instance instead to enabled conversion.")
+		return
+	}
+
+	// Add the spanmetrics connector to the otel config with the converted configuration
+	if otelCfg.Connectors == nil {
+		otelCfg.Connectors = map[otel_component.ID]otel_component.Config{}
+	}
+	otelCfg.Connectors[otel_component.NewID("spanmetrics")] = toSpanmetricsConnector(cfg.SpanMetrics)
+
+	// Add the prometheus exporter to the otel config
+	prometheusID := otel_component.NewID("prometheus")
+	pe := prometheusexporter.NewFactory().CreateDefaultConfig().(*prometheusexporter.Config)
+	if cfg.SpanMetrics.ConstLabels != nil {
+		pe.ConstLabels = *cfg.SpanMetrics.ConstLabels
+	}
+	// Overload the ServerConfig.Endpoint with the MetricsInstance value
+	pe.ServerConfig.Endpoint = "metrics_" + cfg.SpanMetrics.MetricsInstance
+	pe.MetricExpiration = cfg.SpanMetrics.MetricsFlushInterval
+	otelCfg.Exporters[prometheusID] = pe
```

Comment:
Instead of the upstream `prometheusexporter` component, I think we should be converting the custom static mode `prometheusexporter` component which is already in the pipeline. 

This would also help simplify a bit of the logic here since you wouldn't need to do pipeline manipulation anymore to get the other component in there.